Library/ViewModels/SettingsNewslettersCellViewModel.swift

```diff
@@ -0,0 +1,162 @@
+      .map { newsletter, _ in newsletter.displayableName }
+
+    let userAttributeChanged = newsletter
+      .takePairWhen(self.newslettersSwitchTappedProperty.signal.skipNil())
+      .map { newsletter, isOn in
+        (UserAttribute.newsletter(newsletter), isOn)
+      }.logEvents(identifier: "user attribute changed")
+
+    let updatedUser = initialUser
+      .switchMap { user in
+        userAttributeChanged.scan(user) { user, attributeAndOn in
+          let (attribute, on) = attributeAndOn
+          return user |> attribute.lens .~ on
+        }
+    }.logEvents(identifier: "updated user")
+
+    let updateUserAllOn = initialUser
+      .takePairWhen(self.allNewslettersSwitchProperty.signal.skipNil())
+      .map { user, on in
+        return user
+          |> User.lens.newsletters .~ User.NewsletterSubscriptions.all(on: on)
+    }.logEvents(identifier: "updated all")
+
+    let updateEvent = Signal.merge(updatedUser, updateUserAllOn)
+      .switchMap { user in
+        AppEnvironment.current.apiService.updateUserSelf(user)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+    }
+    .logEvents(identifier: "update event")
+
+    self.unableToSaveError = updateEvent.errors()
+      .map { env in
+        env.errorMessages.first ?? Strings.profile_settings_error()
+      }.logEvents(identifier: "Couldn't update user")
+
+    let initialUserOnError = initialUser
+      .takeWhen(self.unableToSaveError)
+      .map { previous in previous }
+
+    self.updateCurrentUser = Signal.merge(initialUser,
+                                          updatedUser,
+                                          updateUserAllOn,
+                                          initialUserOnError)
+      .takeWhen(updateEvent.values().ignoreValues())
+
+    self.subscribeToAllSwitchIsOn = initialUser
+      .map(userIsSubscribedToAll(user:))
+
+    self.switchIsOn = initialUser
+      .combineLatest(with: newsletter)
+      .map(userIsSubscribed(user:newsletter:)).logEvents(identifier: "switchIsOn emitted")
+
+    // Koala
+    userAttributeChanged
+      .observeValues { attribute, on in
+        switch attribute {
+        case let .newsletter(newsletter):
+          AppEnvironment.current.koala.trackChangeNewsletter(
+            newsletterType: newsletter, sendNewsletter: on, project: nil, context: .settings
+          )
+        default: break
+      }
+    }
+  }
+
+  fileprivate let awakeFromNibProperty = MutableProperty(())
+  public func awakeFromNib() {
+    self.awakeFromNibProperty.value = ()
+  }
+
+  fileprivate let initialUserProperty = MutableProperty<User?>(nil)
+  fileprivate let newsletterProperty = MutableProperty<Newsletter?>(nil)
+  public func configureWith(value: (newsletter: Newsletter, user: User)) {
+    self.newsletterProperty.value = value.newsletter
+    self.initialUserProperty.value = value.user
+  }
+  public func configureWith(value: User) {
```

Comment:
Is this used anywhere?

Reply:
Yes. The topCell (`Subscribe to all`) and the other newsletter cells share the same viewModel. This is the function that configures the top cell, since it doesn't require a `Newsletter` for configuration.